test/functional/admin/emergency_banner_controller_test.rb

```diff
@@ -0,0 +1,70 @@
+require "fakeredis/minitest"
+require "test_helper"
+
+class Admin::EmergencyBannerControllerTest < ActionController::TestCase
+  extend Minitest::Spec::DSL
+
+  setup do
+    login_as :gds_admin
+    @controller = Admin::EmergencyBannerController.new
```

Comment:
is this needed?

Reply:
Unfortunately, yes.  As the context blocks obscure the name of the test, minitest doesn't attempt to automatically decide which controller to use.